drivers/sensor/xmc4xxx_temp/xmc4xxx_temp.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2023 Andriy Gelman <andriy.gelman@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT infineon_xmc4xxx_temp
+
+#include <xmc_scu.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/sensor.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(xmc4xxx_temp, CONFIG_SENSOR_LOG_LEVEL);
+
+struct xmc4xxx_temp_data {
+	float temp_out;
+};
+
+static int xmc4xxx_temp_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct xmc4xxx_temp_data *data = dev->data;
+	int32_t val;
+	int ret;
+
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_DIE_TEMP) {
+		return -ENOTSUP;
+	}
+
+	ret = XMC_SCU_StartTemperatureMeasurement();
+	if (ret != 0) {
+		return -EBUSY;
+	}
+
+	while (XMC_SCU_IsTemperatureSensorBusy()) {
```

Comment:
How long does this take? Perhaps insert a sleep to allow other threads to do work?

Reply:
The documentation doesn't actually say. But, I put a counter in the while loop and it reaches ~350 per temperature reading. Also, k_uptime_get() rarely ticks over.  So, I don't think we need to yield here.